sql/expression_value_conversions.cc

```diff
@@ -568,5 +568,541 @@ convertEmbeddingImpl(void * to,
+        initializeBuffer((std::vector<uint8_t> *)buffer, numElements, val);
+#endif
+        return;
+    case ST_STRING:
+        initializeBuffer((std::string *)buffer, numElements, val);
+        return;
+    case ST_UTF8STRING:
+        initializeBuffer((Utf8String *)buffer, numElements, val);
+        return;
+    case ST_ATOM:
+        initializeBuffer((CellValue *)buffer, numElements, val);
+        return;
+    case ST_BOOL:
+        initializeBuffer((bool *)buffer, numElements, val);
+        return;
+    case ST_TIMESTAMP:
+        initializeBuffer((Date *)buffer, numElements, val);
+        return;
+    case ST_TIMEINTERVAL:
+        throw HttpReturnException(500, "Can't store time intervals");
+    }
+    throw HttpReturnException(500, "convertEmbedding unknown time");
+}
+
+template<typename T>
+void defaultConstructBuffer(T * buf, size_t numElements)
+{
+    std::uninitialized_fill(buf, buf + numElements, T());
+}
+
+void initializeStorageBuffer(void * buffer, StorageType storageType,
+                             size_t numElements)
+{
+    switch (storageType) {
+    case ST_FLOAT32:
+        defaultConstructBuffer((float *)buffer, numElements);
+        return;
+    case ST_FLOAT64:
+        defaultConstructBuffer((double *)buffer, numElements);
+        return;
+    case ST_INT8:
+        defaultConstructBuffer((int8_t *)buffer, numElements);
+        return;
+    case ST_UINT8:
+        defaultConstructBuffer((uint8_t *)buffer, numElements);
+        return;
+    case ST_INT16:
+        defaultConstructBuffer((int16_t *)buffer, numElements);
+        return;
+    case ST_UINT16:
+        defaultConstructBuffer((uint16_t *)buffer, numElements);
+        return;
+    case ST_INT32:
+        defaultConstructBuffer((int32_t *)buffer, numElements);
+        return;
+    case ST_UINT32:
+        defaultConstructBuffer((uint32_t *)buffer, numElements);
+        return;
+    case ST_INT64:
+        defaultConstructBuffer((int64_t *)buffer, numElements);
+        return;
+    case ST_UINT64:
+        defaultConstructBuffer((uint64_t *)buffer, numElements);
+        return;
+    case ST_BLOB:
+        // Unfortunately, gcc6.1.1 ICEs when enabling this
+        throw HttpReturnException(600, "Not done: blob conversions");
+#if 0
+        defaultConstructBuffer((std::vector<uint8_t> *)buffer, numElements);
+#endif
+        return;
+    case ST_STRING:
+        defaultConstructBuffer((std::string *)buffer, numElements);
+        return;
+    case ST_UTF8STRING:
+        defaultConstructBuffer((Utf8String *)buffer, numElements);
+        return;
+    case ST_ATOM:
+        defaultConstructBuffer((CellValue *)buffer, numElements);
+        return;
+    case ST_BOOL:
+        defaultConstructBuffer((bool *)buffer, numElements);
+        return;
+    case ST_TIMESTAMP:
+        defaultConstructBuffer((Date *)buffer, numElements);
+        return;
+    case ST_TIMEINTERVAL:
+        throw HttpReturnException(500, "Can't store time intervals");
```

Comment:
We don't have a TimeInterval class that we can use as an underlying type, and it already wasn't supported.  I agree we should address this, but it's out of scope of this ticket.  (MLDB-1988)